_posts/2022-05-14-appsync-graphql-subscriptions.md

```diff
@@ -0,0 +1,308 @@
+
+API that stands for `Application Programming Interface` is a software intermediary that allows two applications to talk to each other. APIs provide a standardized way for two applications to send data back and forth.
+There are many approaches to building an API including the REST APIs. A REST API is an architectural concept for network-based software. When using a REST API to fetch information, you’ll always get back a complete dataset. For example, if you wanted to request information from two sources, you’d need to perform two REST API requests.
+
+##### What is GraphQL?
+
+GraphQL is a query language for your API, a specification, and a set of tools that operates over a single endpoint using HTTP.
+
+GraphQL gives the client the power to ask for exactly what they need and get back predictable results. Making updates to data is made simple with graphql mutations, allowing developers to describe how the data should change.
+
+##### The Core Difference Between REST APIs and GraphQL
+
+GraphQL follows the same set of constraints as REST APIs, but it shifts control over what data is returned (or mutated) to the client.
+Over the last few years, REST has been used to make new APIs, while the focus of GraphQL has been to optimize for performance and flexibility.
+
+##### What is AWS AppSync?
+
+AWS AppSync, I translate it to "API Gateway for GraphQL". It provides a scalable GraphQL interface that allows you to connect your GraphQL schema to multiple data sources like Amazon DynamoDB, AWS Lambda, and HTTP APIs etc.
+
+#### What is a real-time web application?
+
+A traditional application that is not updated in realtime, requires the user to request again to get the latest data. In this model, when user sends the initial request to the server, they have to reload websites/application to see the latest data.
+
+In contrast, a real-time web application updates immediately when new information is available without having to reload or request new data for the page. This is accomplished through a constant two way communication channel that remains open throughout the life of the user session. This constant connection allows the application server to send messages to the client. Applications that use real-time data, needs a mechanism where the server can push an updated response.
+
+In this tutorial, you’ll learn how to bring a realtime functionality into your app by implementing GraphQL subscriptions and AWS directives in your Appsync schema. The goal is to implement a graphql subscription simply with "@aws_subscribe" directive.
```

Comment:
simply with "@aws_subscribe" directive.
should be
simply with the "@aws_subscribe" directive.